internal/action/edit.go

```diff
@@ -46,26 +46,32 @@ func (s *Action) edit(ctx context.Context, c *cli.Context, name string) error {
 		content = bytes.TrimPrefix(content, []byte(secret.Ident+"\n"))
 	}
 
+	// preserve intermediate state across retries
+	tmpContent := content
 	for {
 		// invoke the editor to let the user edit the content
-		nContent, err := editor.Invoke(ctx, ed, content)
+		newContent, err := editor.Invoke(ctx, ed, tmpContent)
 		if err != nil {
 			return ExitError(ExitUnknown, err, "failed to invoke editor: %s", err)
 		}
+		tmpContent = newContent
 		if fromMIME {
-			nContent = append([]byte(secret.Ident+"\n"), nContent...)
-			if _, err := secparse.Parse(nContent); err != nil {
-				cont, err := termio.AskForBool(ctx, "ERROR: Can not parse Gopass native Secret. Retry?", true)
+			newContent = append([]byte(secret.Ident+"\n"), newContent...)
+			if _, err := secparse.Parse(newContent); err != nil {
+				out.Red(ctx, "ERROR: Can not parse Gopass native Secret: %s", err)
+				out.Yellow(ctx, "Hint: Does your Key-Value section end with a new line?")
+				cont, err := termio.AskForBool(ctx, "Retry?", true)
 				if err != nil {
 					return err
 				}
 				if !cont {
 					return fmt.Errorf("malformed secret")
 				}
+				// retain edited content for next try
```

Comment:
this could rather be a comment for line 57, no?

Reply:
Uhm, yes. Indeed. I moved some things around.